File: frontend/src/services/categoryApi.ts
import { fetchApi } from './api';
import { Category, CreateCategoryPayload } from '../types/category';

export const categoryApi = {
  getCategories: () => fetchApi<Category[]>('/categories'),
  getCategory: (id: string) => fetchApi<Category>(`/categories/${id}`),
  createCategory: (payload: CreateCategoryPayload) =>
    fetchApi<Category>('/categories', {
      method: 'POST',
      body: JSON.stringify(payload)
    }),
  updateCategory: (id: string, payload: CreateCategoryPayload) =>
    fetchApi<Category>(`/categories/${id}`, {
      method: 'PUT',
      body: JSON.stringify(payload)
    }),
  deleteCategory: (id: string, reassign_to_category_id?: string) => {
    const query = reassign_to_category_id ? `?reassign_to_category_id=${reassign_to_category_id}` : '';
    return fetchApi<{ detail: string }>(`/categories/${id}${query}`, {
      method: 'DELETE'
    });
  }
};

File: android/app/src/main/kotlin/com/fintrack/app/data/api/CategoryApi.kt
package com.fintrack.app.data.api

import com.fintrack.app.data.models.CategoryCreateRequest
import com.fintrack.app.data.models.CategoryResponse
import com.fintrack.app.data.models.CategoryUpdateRequest
import com.fintrack.app.data.models.MessageResponse
import retrofit2.Response
import retrofit2.http.*

interface CategoryApi {

    @GET("categories")
    suspend fun getCategories(): Response<List<CategoryResponse>>

    @POST("categories")
    suspend fun createCategory(@Body request: CategoryCreateRequest): Response<CategoryResponse>

    @PUT("categories/{id}")
    suspend fun updateCategory(
        @Path("id") id: String,
        @Body request: CategoryUpdateRequest
    ): Response<CategoryResponse>

    @DELETE("categories/{id}")
    suspend fun deleteCategory(@Path("id") id: String): Response<MessageResponse>
}
